docs/apache-airflow/core-concepts/dags.rst

```diff
@@ -484,6 +484,125 @@ You can also combine this with the :ref:`concepts:depends-on-past` functionality
     .. image:: /img/branch_with_trigger.png
 
 
+.. _concepts:setup-and-teardown:
+
+Setup and Teardown
+~~~~~~~~~~~~~~~~~~
+
+In data workflows it's common to create resources (such as a compute resource), use it to do some work, and then tear it down. Airflow provides setup and teardown tasks to support this need.
+
+Key features of setup and teardown tasks:
+
+  * If you clear a task, its setups and teardowns will be cleared.
+  * By default, teardown tasks are ignored for the purpose of evaluating dag run state.
+  * A teardown task will run if it's setup was successful, even if its work tasks failed.
+  * Teardown tasks are ignored when setting dependencies against task groups.
+  * A setup task must always have a teardown and vice versa. You may use EmptyOperator as a setup or teardown.
+
+Basic usage
+"""""""""""
+
+Suppose you have a dag that creates a cluster, runs a query, and deletes the cluster. Without using setup and teardown tasks you might set these relationships:
+
+.. code-block:: python
+
+  create_cluster >> run_query >> delete_cluster
+
+We can use the ``as_teardown`` let airflow know that ``create_cluster`` is a setup task and ``delete_cluster`` is its teardown:
+
+.. code-block:: python
+
+  create_cluster >> run_query >> delete_cluster.as_teardown(setups=create_cluster)
```

Comment:
Is it possible to have multiple setups and teardowns within one scope? (e.g. [s1, s2] >> w1 >> [t1, t2])? If possible, I think it would be helpful to show an example of the syntax here in basic usage. If not possible, would be good to state that they are only one per scope.

Reply:
yes, you can do this kind of thing.  thanks i'll add something.